tests/functional/olp-cpp-sdk-authentication/AuthenticationClientTest.cpp

```diff
@@ -337,13 +335,12 @@ TEST_F(AuthenticationClientTest, SignUpInUser) {
   EXPECT_EQ(olp::http::HttpStatusCode::NO_CONTENT, response4.status);
   EXPECT_STREQ(kErrorNoContent.c_str(), response4.error.c_str());
 
-  // We need to wait for user deletion on server some time before checking
-  // that it is fully deleted.
-  std::this_thread::sleep_for(kDeleteUserWait);
-
   AuthenticationClient::SignInUserResponse response5 = SignInUser(email);
-  EXPECT_EQ(olp::http::HttpStatusCode::UNAUTHORIZED,
-            response5.GetResult().GetStatus())
+  // According to the AAA team, we should expect one of 401 or 404 status.
+  EXPECT_TRUE(olp::http::HttpStatusCode::UNAUTHORIZED ==
+                  response5.GetResult().GetStatus() ||
+              olp::http::HttpStatusCode::NOT_FOUND ==
+                  response5.GetResult().GetStatus())
```

Comment:
Not like this. Use EXPECT_THAT with matchers, at least like that you will get a proper error log:

```cpp
EXPECT_THAT(response5.GetResult().GetStatus(), AnyOf(Eq(UNAUTHORIZED), Eq(NOT_FOUND)));
```